pkg/minishift/registration/registration.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright (C) 2016 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registration
+
+import (
+	"errors"
+	"github.com/docker/machine/libmachine/log"
+	"github.com/docker/machine/libmachine/drivers"
+)
+
+var (
+	ErrDetectionFailed = errors.New("Distro type not recognized to Registration")
+	registrators       = make(map[string]*RegisteredRegistrator)
+	detector  Detector = &StandardRegistrator{}
+)
+
+
+type SSHCommander interface {
+	// Short-hand for accessing an SSH command from the host.
+	SSHCommand(args string) (string, error)
+}
+
+type Detector interface {
+	DetectRegistrator(d drivers.Driver) (Registrator, error)
+}
+
+type StandardRegistrator struct{}
+
+func SetDetector(newDetector Detector) {
+	detector = newDetector
+}
+
+// Registration defines distribution specific actions
+type Registrator interface {
+	SSHCommander
+	// Get the directory where the settings files for docker are to be found
+	GetRegister() error
+
+	// Return the auth options used to configure remote connection for the daemon.
+	GetUnregister() error
+
+	// Figure out if this is the right registration tool to use based on linux distro
+	CompatibleWithDistribution() bool
```

Comment:
@hferentschik this method is just return bool as it implemented like for rhel it will check if `subscription-manager` command available, if someone want to implement for enterprise suse then they can implement this method to check if `smt` command available. For others it just return false and we put a nice message as we discussed `Distro don't support registration` and continue with it.